DataStructure/src/main/list/ResizableArray.java

```diff
@@ -165,4 +191,142 @@ public boolean isEmpty(){
     public int size(){
         return writeIndex;
     }
+
+    /**
+     * add all the elements from given resizableArray
+     * @param resizableArray
+     */
+    public void addAll(ResizableArray resizableArray){
+        if (list.length - size() < resizableArray.size()){
+            list = Arrays.copyOf(list,size()+resizableArray.size());
+        }
+        for (int i = 0; i < resizableArray.size(); i++){
+            add(resizableArray.get(i));
+        }
+    }
+
+    /**
+     * Removes elements which marches with the elements of given resizable array
+     * @param resizableArray
+     */
+    public void removeAll(ResizableArray resizableArray) {
+        for (int i = 0; i < resizableArray.size(); i++) {
+            remove(resizableArray.get(i));
+        }
+        list = Arrays.copyOf(list,writeIndex);
+    }
+
+    /**
+     * retains elements which matches with the elements of given resizable array.
+     * @param resizableArray
+     */
+    public boolean retainAll(ResizableArray resizableArray){
+        if (isEmpty() || resizableArray.isEmpty()){
+            return false;
+        }
+        for (int i = 0; i < size(); i++) {
+            int flag = 0;
+            for (int j = 0; j < resizableArray.size(); j++) {
+                if (list[i] == resizableArray.get(j)) {
+                    flag = 1;
+                    break;
+                }
+            }
+            if (flag == 0){
+                list[i] = -1;
+            }
+        }
+        remove(-1);
+        list = Arrays.copyOf(list,writeIndex);
+        return true;
+    }
+
+    /**
+     * Checks the elements of given resizable array exist in this resizable array
+     * @param resizableArray
+     * @return true when exist otherwise false
+     */
+    public boolean containsAll(ResizableArray resizableArray){
+        for (int i = 0; i < resizableArray.size(); i++) {
+            if (!contains(resizableArray.get(i))){
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Gives array that have element of resizable array
+     * @return array variable
+     */
+    public int[] toArray(){
+        if (isEmpty()){
+            throw new ArrayIndexOutOfBoundsException();
+        }
```

Comment:
Why are you throwing exception ? We an empty resizable array is perfectly valid resizable array.